cmd/gardener-seed-admission-controller/app/gardener_seed_admission_controller.go

```diff
@@ -0,0 +1,242 @@
+	"net/http"
+	"time"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	gardenerlogger "github.com/gardener/gardener/pkg/logger"
+	"github.com/gardener/gardener/pkg/seedadmission"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var logger = gardenerlogger.NewLogger("info")
+
+// Options has all the context and parameters needed to run a Gardener seed admission controller.
+type Options struct {
+	// BindAddress is the address the HTTP server should bind to.
+	BindAddress string
+	// Port is the port that should be opened by the HTTP server.
+	Port int
+	// ServerCertPath is the path to a server certificate.
+	ServerCertPath string
+	// ServerKeyPath is the path to a TLS private key.
+	ServerKeyPath string
+	// Kubeconfig is path to a kubeconfig file. If not given it uses the in-cluster config.
+	Kubeconfig string
+}
+
+// AddFlags adds flags for a specific Scheduler to the specified FlagSet.
+func (o *Options) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&o.BindAddress, "bind-address", "0.0.0.0", "address to bind to")
+	fs.IntVar(&o.Port, "port", 9443, "server port")
+	fs.StringVar(&o.ServerCertPath, "tls-cert-path", "", "path to server certificate")
+	fs.StringVar(&o.ServerKeyPath, "tls-private-key-path", "", "path to client certificate")
+	fs.StringVar(&o.Kubeconfig, "kubeconfig", "", "path to a kubeconfig")
+}
+
+// Validate validates all the required options.
+func (o *Options) validate(args []string) error {
+	if len(o.BindAddress) == 0 {
+		return fmt.Errorf("missing bind address")
+	}
+
+	if o.Port == 0 {
+		return fmt.Errorf("missing port")
+	}
+
+	if len(o.ServerCertPath) == 0 {
+		return fmt.Errorf("missing server tls cert path")
+	}
+
+	if len(o.ServerKeyPath) == 0 {
+		return fmt.Errorf("missing server tls key path")
+	}
+
+	if len(args) != 0 {
+		return errors.New("arguments are not supported")
+	}
+
+	return nil
+}
+
+func (o *Options) run(ctx context.Context) {
+	run(ctx, o.BindAddress, o.Port, o.ServerCertPath, o.ServerKeyPath, o.Kubeconfig)
+}
+
+// NewCommandStartGardenerSeedAdmissionController creates a *cobra.Command object with default parameters
+func NewCommandStartGardenerSeedAdmissionController(ctx context.Context) *cobra.Command {
+	opts := &Options{}
+
+	cmd := &cobra.Command{
+		Use:   "gardener-seed-admission-controller",
+		Short: "Launch the Gardener seed admission controller",
+		Long:  `The Gardener seed admission controller serves a validation webhook endpoint for resources in the seed clusters.`,
+		Run: func(cmd *cobra.Command, args []string) {
+			utilruntime.Must(opts.validate(args))
+			opts.run(ctx)
```

Comment:
WDYT about printing all the flags before starting the controller, like the kubernetes binaries do? (Could also be added to our other binaries). Just for convenience and debugging purposes...

```suggestion
			
			logger.Infof("Starting Gardener seed admission controller...")
			cmd.Flags().VisitAll(func(flag *pflag.Flag) {
				logger.Infof("FLAG: --%s=%s", flag.Name, flag.Value)
			})
			
			opts.run(ctx)
```

This will print out something like:
```
time="2020-03-20T11:10:22+01:00" level=info msg="Starting Gardener seed admission controller..."
time="2020-03-20T11:10:22+01:00" level=info msg="FLAG: --bind-address=0.0.0.0"
time="2020-03-20T11:10:22+01:00" level=info msg="FLAG: --help=false"
time="2020-03-20T11:10:22+01:00" level=info msg="FLAG: --kubeconfig=/Users/d067603/.kube/configs/local/kind-1.yaml"
time="2020-03-20T11:10:22+01:00" level=info msg="FLAG: --port=9443"
time="2020-03-20T11:10:22+01:00" level=info msg="FLAG: --tls-cert-path=example/seed-admission-controller/tls.crt"
time="2020-03-20T11:10:22+01:00" level=info msg="FLAG: --tls-private-key-path=example/seed-admission-controller/tls.key"
```